### PR TITLE
rewamp filterbar

### PR DESCRIPTION
**Pull request by feruzm**
Changes include front page, leaderboard, activities, profile pages!

Rework of filterbar, better visibility and quicker access for users.

Simulator Screen Shot - iPhone 7 Plus - 2019-11-27 at 14 47 30
Simulator Screen Shot - iPhone 7 Plus - 2019-11-27 at 14 47 38
Simulator Screen Shot - iPhone 7 Plus - 2019-11-27 at 14 47 42

### CODE REVIEW

File: src/components/filterBar/view/filterBarStyles.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import EStyleSheet from 'react-native-extended-stylesheet';
+import { relative } from 'path';
```
Comment (ue) [inline src/components/filterBar/view/filterBarStyles.js]: I guess you imported them by mistake?

File: src/components/commentsDisplay/view/commentsDisplayView.js [hunk 2]
```diff
@@ -20,14 +20,15 @@ class CommentsDisplayView extends PureComponent {
...
+      selectedOptionIndex: 0,
     };
   }
 
   // Component Life Cycles
 
   // Component Functions
-  _handleOnDropdownSelect = (index, option) => {
-    this.setState({ selectedFilter: option });
+  _handleOnDropdownSelect = (option, ind) => {
```
Comment (ue) [inline src/components/commentsDisplay/view/commentsDisplayView.js]:
```suggestion
  _handleOnDropdownSelect = (option, index) => {
```

File: src/components/commentsDisplay/view/commentsDisplayView.js [hunk 3]
```diff
@@ -20,14 +20,15 @@ class CommentsDisplayView extends PureComponent {
...
+    this.setState({ selectedFilter: option, selectedOptionIndex: ind });
```
Comment (ue) [inline src/components/commentsDisplay/view/commentsDisplayView.js]:
```suggestion
    this.setState({ selectedFilter: option, selectedOptionIndex: index });
```

**Comment by ue** [comment]
we have already ui componets we creating own ui components therefor we should not import any external ui packages liek `react-native-elements` etc.

File: src/components/filterBar/view/filterBarStyles.js (hunk 1 above)
Comment (feruzm) [reply src/components/filterBar/view/filterBarStyles.js]: yes likely it was already imported, will remove

**Comment by feruzm** [comment]
> we have already ui componets we creating own ui components therefor we should not import any external ui packages liek `react-native-elements` etc.

Do we have buttonGroup or similar component?

File: src/components/notification/view/notificationView.js [hunk 4]
```diff
@@ -45,11 +43,12 @@ class NotificationView extends PureComponent {
   // Component Functions
 
   _handleOnDropdownSelect = async index => {
+    console.log(index);
```
Comment (ue) [inline src/components/notification/view/notificationView.js]: should remove

File: src/constants/options/filters.js [hunk 5]
```diff
@@ -1,5 +1,5 @@
-export const POPULAR_FILTERS = ['TRENDING', 'NEW', 'PROMOTED'];
-export const POPULAR_FILTERS_VALUE = ['trending', 'created', 'promoted'];
+export const POPULAR_FILTERS = ['FEED', 'TOP', 'HOT', 'NEW'];
+export const POPULAR_FILTERS_VALUE = ['feed', 'trending', 'hot', 'created'];
```
Comment (mistikk) [inline src/constants/options/filters.js]: i was use these values for new url scheme feature, you forgot to add `promoted` or you removed that

Comment (mistikk) [reply src/constants/options/filters.js]: also we should handle promoted keyword on search and deep link right? @feruzm

File: src/components/filterBar/view/filterBarView.js [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import { View, Text, TouchableOpacity } from 'react-native';
 import { Icon } from '../../icon';
+import { ButtonGroup } from 'react-native-elements';
```
Comment (mistikk) [inline src/components/filterBar/view/filterBarView.js]: Can we change with our ui elements, we are just using button group but importing all react-native-elements features. We have tags button group, i think we can use that, maybe you cannot find in buttons folder

File: src/constants/options/filters.js (hunk 5 above)
Comment (feruzm) [reply src/constants/options/filters.js]: let's remove promoted @mistikk

File: src/components/filterBar/view/filterBarView.js (hunk 6 above)
Comment (feruzm) [reply src/components/filterBar/view/filterBarView.js]: yes, just now pushed comment with use of Tag component, removed rn-elements

**Comment by feruzm** [comment]
@mistikk and @ue, We only need to themify this and it should be ready...

**Pull request merged**